### PR TITLE
enhancement/17270-flowmap-series-docs-demos

### PR DESCRIPTION
**Pull request by hubertkozik**
Added the API options and demo. Probably I missed some things, so I am waiting for your suggestions.

### CODE REVIEW

File: ts/Series/FlowMap/FlowMapSeries.ts [hunk 1]
```diff
@@ -77,19 +86,16 @@ class FlowMapSeries extends SankeySeries {
...
          * @private
          */
         keys: [
             'from',
             'to',
-            'curveFactor',
-            'weight',
-            'growTowards',
-            'markerEnd'
+            'weight'
```
Comment (raf18seb) [inline ts/Series/FlowMap/FlowMapSeries.ts:98]: If it fits 80, can we make it into one line?

File: ts/Series/FlowMap/FlowMapSeries.ts [hunk 2]
```diff
@@ -489,4 +495,124 @@ export default FlowMapSeries;
...
+ * @type      {number}
+ * @product   highmaps
+ * @apioption series.flowmap.data.curveFactor
+ */
+
+/**
+ * If set to `true`, the line will grow towards its end.
+ *
+ * @sample {highmaps} maps/demo/flowmap-ship-route/
+ *         Example ship route
```
Comment (raf18seb) [inline ts/Series/FlowMap/FlowMapSeries.ts:564]: Misaligned?

File: ts/Series/FlowMap/FlowMapSeries.ts [hunk 3]
```diff
@@ -489,4 +495,124 @@ export default FlowMapSeries;
...
+ * @type      {number}
+ * @product   highmaps
+ * @apioption series.flowmap.data.markerEnd.width
+ */
+
+/**
+ * The weight of the link.
+ *
+ * @sample {highmaps} maps/demo/flowmap-ship-route/
+ *         Example ship route
```
Comment (raf18seb) [inline ts/Series/FlowMap/FlowMapSeries.ts:611]: Misaligned?

**Review by raf18seb — CHANGES_REQUESTED** [review]
Few small comments, see below

File: ts/Series/FlowMap/FlowMapSeries.ts [hunk 4]
```diff
@@ -489,4 +495,124 @@ export default FlowMapSeries;
...
+ * @product   highmaps
+ * @apioption series.flowmap.data
+ */
+
+/**
+ * Higher numbers makes the links more curved. A `curveFactor` of 0 makes the
+ * lines straight.
+ *
+ * @sample {highmaps} maps/demo/flowmap-ship-route/
+ *         Example ship route
```
[the remark this answers is not in the document]
Comment (raf18seb) [reply ts/Series/FlowMap/FlowMapSeries.ts:553]: My mistake, it shouldn't be aligned for `@sample`

File: ts/Series/FlowMap/FlowMapSeries.ts [hunk 5]
```diff
@@ -489,4 +495,124 @@ export default FlowMapSeries;
...
+ * @product   highmaps
+ * @apioption series.flowmap.data.growTowards
+ */
+
+/**
+ * The marker-end defines the arrowhead that will be drawn at the final vertex
+ * of the given path.
+ *
+ * @sample {highmaps} maps/demo/flowmap-ship-route/
+ *         Example ship route
```
[the remark this answers is not in the document]
Comment (raf18seb) [reply ts/Series/FlowMap/FlowMapSeries.ts:576]: My mistake, it shouldn't be aligned for `@sample`

File: ts/Series/FlowMap/FlowMapSeries.ts (hunk 2 above)
Comment (raf18seb) [reply ts/Series/FlowMap/FlowMapSeries.ts:564]: My mistake, it shouldn't be aligned for `@sample`

File: ts/Series/FlowMap/FlowMapSeries.ts (hunk 3 above)
Comment (raf18seb) [reply ts/Series/FlowMap/FlowMapSeries.ts:611]: My mistake, it shouldn't be aligned for `@sample`

**Review by raf18seb — APPROVED** [review]
Thanks!

File: ts/Series/FlowMap/FlowMapSeries.ts [hunk 6]
```diff
@@ -77,20 +86,13 @@ class FlowMapSeries extends SankeySeries {
...
         /**
-         * DESCRIPTION
-         *
          * @declare Highcharts.SeriesFlowMapSeriesDataLabelsOptionsObject
          *
+         * @extends series.mappoint.keys
+         * @default ['from', 'to', 'weight']
          * @private
          */
-        keys: [
```
Comment (madepiet) [inline ts/Series/FlowMap/FlowMapSeries.ts:86]: Why has this been changed here? Shouldn't the default be: 'from', 'to', 'curveFactor', 'weight', 'growTowards', 'markerEnd'

**Review by DJTechnoo — APPROVED** [review]
Looks good to me after resolving the remaining comments.

File: ts/Series/FlowMap/FlowMapSeries.ts (hunk 6 above)
Comment (hubertkozik) [reply ts/Series/FlowMap/FlowMapSeries.ts:86]: @KacperMadej suggested, that `curveFactor`, `growTowards`, `markerEnd` should be set as default from the series object, so it shouldn't be in `keys` property by default.

**Review by madepiet — APPROVED** [review]

**Review by KacperMadej — COMMENTED** [review]
LGTM - one small comment in the code.

File: samples/maps/demo/flowmap-ship-route/demo.js [hunk 7]
```diff
@@ -0,0 +1,176 @@
...
+                lat: 22.39,
+                lon: 114.11
+            }]
+        }, {
+            type: 'flowmap',
+            name: 'Ship Routes',
+            data: [{
+                from: 'Houston',
+                to: 'Miami',
+                weight: 5,
```
[the remark this answers is not in the document]
Comment (hubertkozik) [reply samples/maps/demo/flowmap-ship-route/demo.js:89]: Yes, I assumed `weight = days of travel`. I will add info to the subtitle.

**Review by KacperMadej — APPROVED** [review]

**Pull request merged**